test/resources/mnist/mnist_script_mode.py

```diff
@@ -114,13 +117,15 @@ def save(model_dir, model):
     parser.add_argument('--train', type=str, default=os.environ['SM_CHANNEL_TRAIN'])
     parser.add_argument('--test', type=str, default=os.environ['SM_CHANNEL_TEST'])
 
-    parser.add_argument('--hosts', type=str, default=os.environ['SM_HOSTS'])
     parser.add_argument('--current-host', type=str, default=os.environ['SM_CURRENT_HOST'])
-
-    num_gpus = int(os.environ['SM_NUM_GPUS'])
+    parser.add_argument('--hosts', type=str, default=os.environ['SM_HOSTS'])
 
     args = parser.parse_args()
 
-    model = train(args.batch_size, args.epochs, args.learning_rate, num_gpus, args.train, args.test,
-                  ast.literal_eval(args.hosts), args.current_host)
-    save(args.model_dir, model)
+    num_gpus = int(os.environ['SM_NUM_GPUS'])
+    hosts = ast.literal_eval(args.hosts)
```

Comment:
remove ast.literal_eval. the env var default is supposed to be json so we should only parse it as json (and throw error if it isn't). and user entering command line should be able to enter repeated values not handwritten json.

Reply:
fixed